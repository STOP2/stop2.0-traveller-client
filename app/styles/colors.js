const colors = {
    white: 'white',
    black: 'black',
    transparent: 'transparent',
    busBlue: '#007AC9',
    tramGreen: '#00985F',
    trainViolet: '#8C4799',
    metroOrange: '#FF6319',
    ferryBlue: '#00B9E4',
    lightFerryBlue: '#BEE4F8',
    HSLpink: '#F092CD',
    HSLlightPink: '#F4DEEA',
    HSLgreen: '#64BE14',
    HSLlightGreen: '#D0E6BD',
    HSLalarmRed: '#DC0451',
    HSLyellow: '#FED100',
    K80: '#333333',
    CoolGray: '#616365',
    K40: '#666666',
    K10: '#dddddc',
    stopBackground: '#e9e9e9',
    tryAgain: '#0000ff',
    statusBar: '#006AAF',
}

export default colors
