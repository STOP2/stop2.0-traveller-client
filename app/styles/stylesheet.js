import { StyleSheet } from 'react-native';

var styles = StyleSheet.create({
  busrow: {
    flexDirection: 'row',
    borderBottomColor:666666,
    borderBottomWidth:1,
    height: 60,
    padding: 10,
    alignItems: 'center'
  },
  busrowheader: {
    flexDirection: 'row',
    padding:10,
    backgroundColor:'#e9e9e9'
  },
  title: {
    fontSize:29,
    backgroundColor:'#007AC9',
    color:'white',
    padding:10
  },
  start: {
    flex:1,
    padding: 10,
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: '#dddddc'
  },
  startText: {
    flex:1,
    color: '#ffffff',
    fontSize: 60,
    fontFamily: 'gotham-rounded-medium'
  },
  locationErrorText: {
    flex:1,
    color: '#007AC9',
    fontSize: 20,
    fontFamily: 'gotham-rounded-book'
  },
  gettingLocationText: {
    flex:1,
    color: '#007AC9',
    fontSize: 30,
    fontFamily: 'gotham-rounded-book'
  },
  stopRequestStyle: {
  flex: 1,
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center'
  },
  spinnerContainer: {
    backgroundColor: 'transparent',
    alignSelf: 'center'
  },
  button: {
    marginBottom: 7,
    padding: 5,
    backgroundColor: '#64BE14',
    borderRadius: 5,
    margin: 20
  },
  confirmText: {
    marginBottom: 7,
    padding: 5,
    color: '#64BE14',
    fontSize: 40,
    textAlign: 'center',
    margin: 20
  },
  startImage: {
    width: 100,
    height: 100,
  },
  startUpper: {
    flex: 1,
    flexDirection:'column',
    alignItems:'center',
    justifyContent:'center',
    backgroundColor: '#00B9E4',
    borderRadius: 10,
    marginBottom: 3
  },
  startLower: {
    flex: 1,
    flexDirection:'column',
    alignItems:'center',
    justifyContent:'center',
    backgroundColor: '#F092CD',
    borderRadius: 10,
    marginTop: 3
  }
});

export default styles;
