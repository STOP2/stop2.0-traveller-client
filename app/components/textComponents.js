import React, { Component } from 'react'
import { Text } from 'react-native'

import styles from '../styles/stylesheet'

export class DefaultText extends Component {
    render()
    {
        return (
          <Text style={this.props.style}>
            <Text style={styles.defaultText}>{this.props.children}</Text>
          </Text>
        )
    }
}

DefaultText.propTypes = {
    children: React.PropTypes.node,
    style: React.PropTypes.number
}
