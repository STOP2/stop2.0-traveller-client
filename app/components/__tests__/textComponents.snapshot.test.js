import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { DefaultText, BoldText } from '../textComponents';

import styles from '../../styles/stylesheet';

jest.mock('react-native-localization');

describe('DefaultText component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DefaultText style={styles.padding10}>x</DefaultText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('BoldText component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BoldText style={styles.padding10}>x</BoldText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
