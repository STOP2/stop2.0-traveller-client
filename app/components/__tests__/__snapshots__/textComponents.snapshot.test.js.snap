exports[`BoldText component renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "gotham-rounded-medium",
      },
      Object {
        "padding": 10,
      },
    ]
  }>
  x
</Text>
`;

exports[`DefaultText component renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "gotham-rounded-book",
      },
      Object {
        "padding": 10,
      },
    ]
  }>
  x
</Text>
`;
